forem/forem
# Reconfigure session store

__NOTE: ~~This PR is a follow-up to https://github.com/forem/forem/pull/15944 which needs to be merged first~~. I may also just make this part of the Rails 7 update, tbd.__

## What type of PR is this? (check all applicable)


## Description

Since December 2020 we have `redis-actionpack` and `redis-rack` pinned to specific versions in our `Gemfile` (comment updated recently):

```ruby
# NOTE: [@rhymes, @citizen428]: sign in doesn't work with redis-rack 2.1.0+ or with redis-actionpack 5.2.0+
# We need to investigate why exactly, in the meantime we are hardcoding these two dependencies
# to keep them at the latest working version
gem "redis-actionpack", "5.1.0" # Redis session store for ActionPack. Used for storing the Rails session in Redis.
gem "redis-rack", "2.0.6" # Redis Store for Rack applications
```

However, this prevents us from upgrading to Rails 7 which requires a newer version of `redis-actionpack`. After a lot of reading and trial and error I identified one potential way forward:

* completely remove `redis-actionpack` and `redis-rack`
* reconfigure the session store to use `:cache_store` which uses the built-in `:redis_cache_store`, not the `redis_store` provided by `redis-actionpack`

The main drawback with this approach is that it will probably render all current sessions invalid, which is not something we may want:

* Check out `main`
* Start a Rails server, go to localhost:3000 and log in
* Stop the Rails server
* Check out this branch
* Start a Rails server, go to localhost:3000, reload the page, you're no longer logged in

However, given that this approach removes several gems and makes our config a bit simpler to understand it may still be worth it.

Not changing anything related to the gems and removing `signed: true` from the session store configuration also seems to work but seems less than ideal too.

## Related Tickets & Documents

https://github.com/forem/forem/pull/15944

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
